mesonbuild/coredata.py

```diff
@@ -662,11 +678,13 @@ def set_default_options(self, default_options, subproject, env):
                 env.cmd_line_options.setdefault(k, v)
 
         # Set default options as if they were passed to the command line.
-        # Subprojects can only define default for user options.
+        # Subprojects can only define default for user options and not yielding
+        # builtin option.
         from . import optinterpreter
         for k, v in default_options.items():
             if subproject:
-                if optinterpreter.is_invalid_name(k, log=False):
+                if (k not in builtin_options or builtin_options[k].yielding) \
+                      and optinterpreter.is_invalid_name(k, log=False):
```

Comment:
__[Flake8]__ [E126] continuation line over-indented for hanging indent ([link](https://sider.review/gh/repos/19784232/pulls/4157#issue-4358606))

<sub>posted by Sider</sub>